fuse/eval/metrics/sequence_gen/metrics_seq_gen_common.py

```diff
@@ -177,18 +194,18 @@ def _perplexity_update(
     preds = preds.detach()
     target = target.detach()
 
-    preds = preds.reshape(-1, preds.shape[-1])
-    target = target.reshape(-1)
+    preds = preds.view(-1, preds.shape[-1])
```

Comment:
why? to save memory?

Reply:
@mosheraboh From the documentation of [reshape](https://pytorch.org/docs/stable/generated/torch.reshape.html#torch.reshape): When possible, the returned tensor will be a **view of input. Otherwise, it will be a copy**. Contiguous inputs and inputs with compatible strides can be reshaped without copying, but you should not depend on the copying vs. viewing behavior.

Question is: 
(1) do we assume view will always work, and otherwise alert the user. Or 
(2) do we want to be permissive and copy the data (despite additional GPU memory) when view is not possible (i.e. memory is not contiguous)

Reply:
Is it really that expensive in memory?

If it is, than I would warn the user conditionally to `Tensor.is_contiguous()` result :)